pkg/apis/ext.cattle.io/v1/types.go

```diff
@@ -26,3 +27,147 @@ type UserActivityStatus struct {
+	// +optional
+	Description string `json:"description, omitempty"`
+	// TTL is the time-to-live of the token, in milliseconds.
+	// The default, 30 days, is indicated by the value `0`.
+	// +optional
+	TTL int64 `json:"ttl"`
+	// Enabled indicates an active token. The default (`null`) indicates an
+	// enabled token.
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+}
+
+// TokenPrincipal contains the data about the user principal owning the token.
+type TokenPrincipal struct {
+	// Name is the name unique to the authentication provider.
+	Name string `json:"name"`
+	// DisplayName is the human readable name/description of the principal.
+	// +optional
+	DisplayName string `json:"displayName,omitempty"`
+	// LoginName is the account name of the principal in the managing auth provider.
+	LoginName string `json:"loginName,omitempty"`
+	// ProfilePicture is an url to a picture to use when displaying the principal.
+	// +optional
+	ProfilePicture string `json:"profilePicture,omitempty"`
+	// ProfileURL is not used by the system
+	// +optional
+	ProfileURL string `json:"profileURL,omitempty"`
+	// PrincipalType is the kind of principal. Legal values are "user" and "group".
+	PrincipalType string `json:"principalType,omitempty"`
+	// Me is a virtual flag for use with the dashboard.
+	Me bool `json:"me,omitempty"`
+	// MemberOf is a virtual flag for use with the dashboard.
+	MemberOf bool `json:"memberOf,omitempty"`
+	// Provider is the name of the auth provider managing the principal
+	Provider string `json:"provider,omitempty"`
+	// ExtraInfo contains additional information about the principal.
+	ExtraInfo map[string]string `json:"extraInfo,omitempty"`
+}
+
+// TokenStatus defines the most recently observed status of the Token.
+type TokenStatus struct {
+	// Value is the access key. It is shown only on token creation and not saved.
+	Value string `json:"tokenValue,omitempty"`
+	// Hash is the hash of the Value.
+	Hash string `json:"tokenHash,omitempty"`
+	// Current indicates whether the token was used to authenticate the current request.
+	Current bool `json:"current"`
+	// Expired indicates whether the token has exceeded its TTL.
+	Expired bool `json:"expired"`
+	// ExpiresAt is the token's expiration timestamp or an empty string if the token doesn't expire.
+	ExpiresAt string `json:"expiresAt"`
+	// LastUpdateTime is the timestamp of the last change to the token.
+	LastUpdateTime string `json:"lastUpdateTime"`
+	// LastUsedAt is the timestamp of the last time the token was used to authenticate.
+	LastUsedAt *metav1.Time `json:"lastUsedAt,omitempty"`
+}
+
+// Implement the TokenAccessor interface
+
+func (t *Token) GetName() string {
+	return t.Name
+}
+
+func (t *Token) GetIsEnabled() bool {
+	return t.Spec.Enabled == nil || *t.Spec.Enabled
+}
+
+func (t *Token) GetIsDerived() bool {
+	// session is the kind of login tokens, the only kind of non-derived tokens.
+	return t.Spec.Kind != "session"
+}
```

Comment:
`IsEnabled()` or even `Enabled()` look cleaner than `GetIsEnabled()`. Same goes for the `GetIsDerived()`. Wdyt?

Reply:
If we remove `Get` here,  we should remove `Get` from all other interface methods as well.

Reply:
I will remove `Get` from all which have it.

Reply:
Canceled. Without the prefix method and field names are in conflict. And the reason for adding the prefix in the first place.